include/pybind11/pytypes.h

```diff
@@ -491,7 +491,7 @@ class tuple : public object {
         if (!m_ptr) pybind11_fail("Could not allocate tuple object!");
     }
     size_t size() const { return (size_t) PyTuple_Size(m_ptr); }
-    detail::tuple_accessor operator[](size_t index) const { return detail::tuple_accessor(ptr(), index); }
```

Comment:
They look great.  I confess that I didn't review all the places thoroughly.  The only place I was sure was `tuple::operator[]()`, because I traced around it several times using a debugger.

I am trying to use pybind11 to replace a Cython wrapper for my project.  If I find other places to change I will file new PRs.

Thanks a lot for the great project.